src/EnergyPlus/UnitarySystem.cc

```diff
@@ -3218,7 +3252,7 @@ namespace UnitarySystems {
         }
 
         // Early calls to ATMixer don't have enough info to pass GetInput. Need to get the data next time through.
-        if (sysNum == -1 || !state.dataZoneEquip->ZoneEquipInputsFilled) {
+        if (sysNum == -1 && !state.dataZoneEquip->ZoneEquipInputsFilled) {
```

Comment:
!state.dataZoneEquip->ZoneEquipInputsFilled means it is too early for zone equipment or maybe even air loop equipment with a control zone that needs zone node data. The old logic said to return anytime a new instance of the UnitarySystem exists when if all the data is available data gathering should continue. I think sysNum == -1 isn't even needed now but too chicken to remove it.